SmartDeviceLink-iOS/SmartDeviceLink/SDLRPCRequestFactory.m

```diff
@@ -360,15 +368,28 @@ + (SDLRegisterAppInterface *)buildRegisterAppInterfaceWithAppName:(NSString *)ap
     msg.languageDesired = languageDesired;
     msg.hmiDisplayLanguageDesired = hmiDisplayLanguageDesired;
     msg.appID = appID;
-
+    msg.deviceInfo = [self buildDeviceInfo];
     msg.correlationID = [NSNumber numberWithInt:1];
-
-    return msg;
+	return msg;
 }
 
-+ (SDLRegisterAppInterface *)buildRegisterAppInterfaceWithAppName:(NSString *)appName isMediaApp:(NSNumber *)isMediaApp languageDesired:(SDLLanguage *)languageDesired appID:(NSString *)appID {
-    NSMutableArray *syns = [NSMutableArray arrayWithObject:appName];
++(SDLDeviceInfo*) buildDeviceInfo {
+    SDLDeviceInfo *deviceInfo = [[SDLDeviceInfo alloc]init];
+    deviceInfo.hardware = [[UIDevice currentDevice] model];
+    deviceInfo.os = [[UIDevice currentDevice]systemName];
+    deviceInfo.osVersion = [[UIDevice currentDevice]systemVersion];
+    
+    CTTelephonyNetworkInfo *netinfo = [[CTTelephonyNetworkInfo alloc] init];
+    CTCarrier *carrier = [netinfo subscriberCellularProvider];
```

Comment:
I believe it should be covered by the general SDL/AppLink prompt that comes up before you use any mobile app with Sync.